src/applications/personalization/account/components/ConnectedAccountsSection.jsx

```diff
@@ -10,25 +12,60 @@ function recordAnalyticEvent() {
   });
 }
 
-export default function ConnectedAccountsSection() {
-  return (
-    <div>
-      <h3>Connected accounts</h3>
+function ConnectedAccountsSection({
+  accounts,
+  loadConnectedAccounts,
+  loading,
+}) {
+  // TODO: for some reason this hook is called twice on load. maybe
+  // AccountMain's render is to blame because it renders an instance of this
+  // component and then re-renders, replacing it?
+  useEffect(
+    () => {
+      loadConnectedAccounts();
+    },
+    [loadConnectedAccounts],
```

Comment:
Re: [`useEffect`](https://reactjs.org/docs/hooks-reference.html#useeffect)

>By default, effects run after every completed render, but you can choose to fire it only when certain values have changed.

Doesn't look like it's just like `componentDidMount` if it's running after every render. This looks like a new type of lifecycle event (`render` updates the virtual Dom, `componentDidMount` signals the initial markup writing to the DOM, and `useEffect` is called every time the DOM is updated)?

Looks like you're passing the second argument to conditionally call the effect but I think that second argument is used in an identity check. `loadConnectedAccounts` is a function? maybe you want to pass a value like `accounts`?

Reply:
the way I understand this is that if the value of `loadConnectedAccounts` doesn't change, then it won't re-render. So this should function just like `componentDidMount`. I also tried passing an empty array as the second arg, which is the standard way to mimic `componentDidMount`, but that also results in getting fired twice. Also, eslint complains about passing the empty array which is why I pass `loadConnectedAccounts`. So I think something else is up...

Reply:
Might be able to use the react dev tools to dig into it. I would comment [this](https://github.com/department-of-veterans-affairs/vets-website/blob/3dddd2b66f3d6980799ec22d26b2b6a04f8550a8/src/applications/personalization/account/components/AccountMain.jsx#L19) out and see if it keeps happening. There's no `componentShouldUpdate` in the `<AccountMain/>` component so it's going to re-render every time the props / state change but I don't think that would force a re-mount. Another place to look is the effect of that callback on `<AccountApp />` - maybe it's triggering a remount in `<DowntimeNotification />` or `<RequiredLoginView />`

Reply:
Thanks, Rian. Disabling the call to `fetchMHVAccount` was the first step to figuring out the root of this issue. Turns out it was really pretty simple. The initial value for `mhvAccount.loading` was set to `false`, which then got flipped to `true` when `fetchMHVAccount` was called, and finally went back to `false` after the call to get MHV account completed. When that value is `false` it triggered a mount of `<ConnectedAccountsSection />`: https://github.com/department-of-veterans-affairs/vets-website/blob/979226fc44acd9be79fefa10379dfc197b7fd8e7/src/applications/personalization/account/components/AccountMain.jsx#L69-L139